dev/pre-commit.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+"""
+This script can be used as a pre-commit hook when contributing to this project.
+It runs a subset of CI checks that is sufficiently fast to not be annoying when run
+for every commit.
+
+To install, make sure isort and black are installed on your system, then run:
+
+$ ln -sv ../../dev/pre-commit.py ./.git/hooks/pre-commit
```

Comment:
Probably, it would be good to mention this in Developer guide.

Reply:
I added a mention in `CONTRIBUTING.md`, since that's where the coding style is described.